### DISCUSSION TITLE
3.0.0 Release Notes

### DISCUSSION THREAD

**Discussion by Jeffser** [Jeffser/Alpaca#401] [Announcements]
Hi everyone, to celebrate (almost) 500 stars on GitHub, 60k downloads in Flathub and of course the holidays, I prepared this fire update!!!

This update will be published once all translations are completed or the 23rd, whatever comes first.

# System Messages

Probably one of the most requested features is finally here!

System messages are taken as literal instructions by all models, this is a quick way of changing the behavior of models in each chat.

![image](https://github.com/user-attachments/assets/625468b0-c4c9-4996-9710-0d34915fa84c)

To send a system message just write whatever you want and even attach documents, then long press / right click the send button, this will show this nice menu.

![image](https://github.com/user-attachments/assets/657bf189-1346-49e3-92de-5ca0c4ca5dc9)

# Message Redesign

You probably noticed this in the screenies before, yep the messages have been completely redesigned!

The user messages can now be even smaller, the footer is now dimmer and the action buttons have been replaced by the options button, I think I did a great job with this haha.

![image](https://github.com/user-attachments/assets/78050cf4-f078-4c9d-8bd1-e7a3c3d28dd4)

![image](https://github.com/user-attachments/assets/c585803c-33a5-4b2d-bf60-4654d42c4810)

# Editable Model Messages

Did the model made some mistake in the code you want to run? Want to modify the history of chats without having to regenerate messages? Well now with the redesign I won't need to worry about model messages having 4 whole buttons, so yeah, now you can edit model messages.

![image](https://github.com/user-attachments/assets/b9908b7a-b015-4de1-808c-3b4b270ef21d)

# Markdown Exporting

I'm pretty sure no one has ever exported a chat before, there really isn't any reason to do so, but now when you do it the tar file will include a MD file which you can use to represent a chat in a different document app like Obsidian.

![image](https://github.com/user-attachments/assets/22f357c4-782b-4a25-8320-1c80012eb4ee)

# Reload Button in Model Manager

When using a remote instance you might need to update the list of downloaded models, so when you use a remote instance this nice button will appear, it does the thing, pretty simple.

![image](https://github.com/user-attachments/assets/36b93c60-dd4e-4ccd-bd0e-4a06a7db040b)

# ROCm Fix

Tired of seeing this red text? Well with the new Ollama version it will (probably) work fine!

![image](https://github.com/user-attachments/assets/efc07c76-0c2d-41e1-818f-dc68bf41d331)

# Model Directory

Now you can change the directory where models will be stored for the integrated Ollama instance

![image](https://github.com/user-attachments/assets/bba0761d-2100-4d0e-9b14-0908a382e641)

# Alpaca Without Instance

It has been possible to use Alpaca without the integrated instance for a while, but now Alpaca fully adapts to the lack of an integrated instance by hiding the instance options and making it easier to connect to an instance!

With this now I'm able to provide Flatpak and Snap packages without the integrated instance from the releases page!

![image](https://github.com/user-attachments/assets/d75ab9ee-a2bf-44ac-bb1d-23bd2c83dd9b)

# ODT Files Attachments

ODT are mostly used by LibreOffice Writer, Alpaca can now extract titles, paragraphs and tables from them!

![image](https://github.com/user-attachments/assets/b5e81c34-7f95-4599-a2e0-7026270495cc)

# File Previewer Dynamic Size

The file previewer can now resize itself to the content's size.

![image](https://github.com/user-attachments/assets/0d66970b-7d62-46c1-be83-633bb332f352)

# Working Snap Package

I kinda abandoned and stopped testing the Snap version of Alpaca because of Snapcraft not letting me publish the app officially in the Snap Store (it still doesn't) but now that we've reached 3.0.0 I wanted to go back and fix up that package!

There's stuff that still don't work, mainly ROCm and Gnome Search Integration, but that's something I could work on later once they actually let me publish the app officially.

![Screenshot from 2024-12-18 23-40-14](https://github.com/user-attachments/assets/d45c616a-41d0-4b6c-ab19-730dd8323ff7)

# MacOS Release (Beta)

After some work the past few days I managed to package Alpaca for MacOS!

This is just the first version, so there are some problems:

- Font looks weird
- It doesn't open sometimes
- Notifications don't use Alpaca icon
- Global menu doesn't have options and shows title 'Python'

![398285864-cc9a1329-cd7d-4b05-8f73-6f5c2422340a](https://github.com/user-attachments/assets/a3b02f11-4f6c-44c9-b307-5d24b5053df3)

----

# Recommended Models

Before closing this announcement I wanted to recommend some of my favorite models!

- **Llama3.2 Vision:** Surprisingly good image recognition
- **Orca Mini:** Fast and small model, good for spellchecking and simple questions
- **Gemma2:** Great overall model
- **Starcoder2:** Probably the best code model I've tested
- **Exaone3.5:** Haven't tested it but it's kinda funny to have a model made by the guys that made my tv (LG)
- **Tulu3:** Fully open-source data, code and recipes, cool.

I recommend you guys check out the [Ollama Library](https://ollama.com/library), there are a lot of specialized models and stuff.

----

As always I wanted to thank all of you for the support, I didn't expect to end the year with such a great app to call my own. Of course I also thank all of the contributors including people who just made issues with recommendations or reports, translators and code contributors. You guys are awesome.

When I first turned on my sponsors page I didn't expect people to actually give me money for this project, don't give me wrong, I'm really thankful with all of you that gave me money before and also the monthly sponsors, all of you rock. Even if it's just a dollar or a dozen, I'm really honored to have made something that people have found actual worth in.

I'm really glad that I was able to made a bit of an impact in the Linux desktop community, I hope next year comes with a lot of new ideas and projects! 

Who knows, maybe one day I will actually make the distro I've always wanted to have lol.

This was my first year actually participating in open source development, this community is really something special, thanks for always being awesome.

**Comment by luyu-wu** [comment]
Happy holidays and thanks for the great update!

**Comment by olumolu** [comment]
Congratulations for 500star on github @Jeffser 👍🎄

**Comment by VernonHughes-OR** [comment]
Congrats dude! Rootin for ya!
Being able to move where the modeled is stored is awesome! I actually reinstalled because the models were being stored in my root partition, and I didn't care to make a symlink and complicate things for future LTS releases for my distro.

ODT support is also huge dude, that's incredibly useful! Markdown exporting is also another slam dunk! You've been working hard and it shows!

Obviously my amd card working is my favorite fix, thanks for working hard so we don't have to!

**Reply by Jeffser** [reply]
Thanks for the support, I hope the AMD fix finally works properly, there's still a lot of testing to be done before the release haha

**Comment by ghost** [comment]
I just wanna say thank you! What a great xmas gift, I will test Alpaca soon again! 
Thank you so much for your work. <3

**Reply by Jeffser** [reply]
Thanks, I'm glad people enjoy my work <3

**Comment by olumolu** [comment]
![Screenshot_20241225-095044~2](https://github.com/user-attachments/assets/0c59c341-d025-452e-9de2-70807a3c7a1b)
Mentioned in ollama official readme.

**Reply by Jeffser** [reply]
Niceeee

**Reply by olumolu** [reply]
Yeooo https://github.com/ollama/ollama/commit/51af455f624598deee3960d3e9b1837225a97b0b
